examples/nodes/imagenet_benchmark.py

```diff
@@ -0,0 +1,211 @@
+import argparse
+
+import os
+import time
+from typing import Any, Iterator
+
+import torch.utils.data
+import torchdata.nodes as tn
+
+import torchvision.transforms.functional as F
+from PIL import Image
+from torch.utils.data import default_collate
+
+
+class ImagenetTransform:
+    """Decode, transform, and crop to 224x224.
+    If called with a list of dicts, collates the results.
+    """
+
+    def __call__(self, data):
+        if isinstance(data, list):
+            return default_collate([self.transform_one(x) for x in data])
+        else:
+            return self.transform_one(data)
+
+    def transform_one(self, data):
+        img = Image.open(data["img_path"]).convert("RGB")
+        img_tensor = F.pil_to_tensor(img)
+        img_tensor = F.center_crop(img_tensor, [224, 224])
+        data["img"] = img_tensor
+        return data
+
+
+class ImagenetLister:
+    """Access imagenet data through either __getitem__, or an iterator.
+    If using an iterator, will loop forever, in order
+    """
+
+    def __init__(self, path: str):
+        self.path = path
+        self.img_labels = []
+        self.img_paths = []
+        for label in os.listdir(path):
+            for img_path in os.listdir(os.path.join(path, label)):
+                self.img_labels.append(label)
+                self.img_paths.append(os.path.join(path, label, img_path))
+
+        assert len(self.img_labels) == len(self.img_paths), (
+            len(self.img_labels),
+            len(self.img_paths),
+        )
+
+    def __getitem__(self, i: int) -> dict:
+        data = {"img_path": self.img_paths[i]}
+        return data
+
+    def __len__(self):
+        return len(self.img_labels)
+
+    def __iter__(self) -> Iterator[dict]:
+        while True:  # Loop forever
+            for i in range(len(self.img_labels)):
+                yield {"img_path": self.img_paths[i]}
+
+
+class ImagenetDataset(torch.utils.data.Dataset):
+    """Classic DataLoader v1-style dataset (map style). Applies ImagenetTransform when
+    retrieving items.
+    """
+
+    def __init__(self, path: str):
+        self.imagenet_data = ImagenetLister(path)
+        self.tx = ImagenetTransform()
+
+    def __len__(self):
+        return len(self.imagenet_data)
+
+    def __getitem__(self, i: int) -> dict:
+        return self.tx(self.imagenet_data[i])
+
+
+def setup_classic(args):
+    dataset = ImagenetDataset(args.imagenet_dir)
+    assert args.in_order is False, "torch.utils.data.DataLoader does not support out-of-order iteration yet!"
```

Comment:
should the assert be args.in_order is true ? also, does the string have to change with https://github.com/pytorch/pytorch/pull/141833 ?

Reply:
We can drop this after the next release of pytorch, and/or do a version check